### PR TITLE
Test: Addition of tests for replicationQueue publish and publish to dlq

### PR DESCRIPTION
**Pull request by abhishekj720**
**What changed?**
Addition of test for replication queue methods for publishing to DLQ. Includes several edge cases.

**Why?**
To make the code more secured, reliable and understandable. As well to increase test code coverage.

**How did you test it?**
unit tests

No potential risks as it contains only test cases.

### CODE REVIEW

**Review by 3vilhamster — APPROVED** [review]

File: common/domain/replication_queue_test.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	"github.com/uber/cadence/common/persistence"
+	"github.com/uber/cadence/common/types"
+)
+
+func TestReplicationQueueImpl_Publish(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockQueue := persistence.NewMockQueueManager(ctrl)
+	rq := NewReplicationQueue(mockQueue, "testCluster", nil, nil) // Mock other dependencies as needed
```
Comment (taylanisikdemir) [inline common/domain/replication_queue_test.go:40]: let's create a mock object separate per test case. reusing the same object for different cases can cause hard-to-troubleshoot kind of failures.

File: common/domain/replication_queue_test.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/uber/cadence/common/persistence"
+	"github.com/uber/cadence/common/types"
+)
+
+func TestReplicationQueueImpl_Publish(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
```
Comment (taylanisikdemir) [inline common/domain/replication_queue_test.go:37]: this defer not needed.

**Review by taylanisikdemir — APPROVED** [review]
